
From: Chris Wright <chrisw@osdl.org>

Add simple helper function to grab a reference to a user_struct.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/include/linux/sched.h |    5 +++++
 1 files changed, 5 insertions(+)

diff -puN include/linux/sched.h~rlim-add-simple-get_uid-helper include/linux/sched.h
--- 25/include/linux/sched.h~rlim-add-simple-get_uid-helper	Mon Jun  7 14:47:48 2004
+++ 25-akpm/include/linux/sched.h	Mon Jun  7 14:47:48 2004
@@ -756,6 +756,11 @@ extern void __set_special_pids(pid_t ses
 
 /* per-UID process charging. */
 extern struct user_struct * alloc_uid(uid_t);
+static inline struct user_struct *get_uid(struct user_struct *u)
+{
+	atomic_inc(&u->__count);
+	return u;
+}
 extern void free_uid(struct user_struct *);
 extern void switch_uid(struct user_struct *);
 
_
